### PR TITLE
TEST-#5991: add simple tests for `read_orc`, `read_spss`, `json_normalize`, `read_xml`, `read_gbq`

### LINKED ISSUE #5991: add tests for `read_orc`, `read_spss`, `json_normalize`, `read_xml`, `read_gbq`

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5991 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/test/test_io.py [hunk 1]
```diff
@@ -2613,6 +2629,56 @@ def test_to_pickle(self):
...
+ <row>
+   <shape>circle</shape>
+   <degrees>360</degrees>
+   <sides/>
+ </row>
+"""
+        eval_io("read_xml", path_or_buffer=data)
+
+
+class TestOrc:
```
Comment (anmyachev) [inline modin/pandas/test/test_io.py:2651]: For some formats (`orc`, `spss`), it is not easy to add infrastructure. In cases where we default to pandas, I think it's enough to check that the parameters are passed to pandas.

File: modin/pandas/io.py [hunk 2]
```diff
@@ -761,7 +761,7 @@ def read_orc(
     Load an ORC object from the file path, returning a DataFrame.
     """
     ErrorMessage.default_to_pandas("read_orc")
-    return DataFrame(pandas.read_orc(path, columns, **kwargs))
+    return DataFrame(pandas.read_orc(path, columns=columns, **kwargs))
```
Comment (anmyachev) [inline modin/pandas/io.py:764]: This is not a mistake, it's just more convenient to test, from the point of view of logic, nothing changes.

File: modin/core/io/io.py [hunk 3]
```diff
@@ -174,42 +174,9 @@ def read_json(
         summary="Load data from Google BigQuery into query compiler",
         returns=_doc_returns_qc,
     )
-    def read_gbq(
```
Comment (anmyachev) [inline modin/core/io/io.py:177]: Some parameters are outdated (for example `private_key`). I guess we can just use `**kwargs`.

File: modin/pandas/io.py [hunk 4]
```diff
@@ -716,17 +716,17 @@ def to_pickle(
...
 ):  # noqa: PR01, RT01, D200
     """
     Load an SPSS file from the file path, returning a DataFrame.
     """
+    _, _, _, kwargs = inspect.getargvalues(inspect.currentframe())
+
     from modin.core.execution.dispatching.factories.dispatcher import FactoryDispatcher
 
-    return DataFrame(
-        query_compiler=FactoryDispatcher.read_spss(path, usecols, convert_categoricals)
```
Comment (anmyachev) [inline modin/pandas/io.py:728]: It didn't work at all before. Parameters must be passed as keywords.

File: modin/core/io/io.py [hunk 5]
```diff
@@ -174,42 +174,9 @@ def read_json(
...
-        location=None,
-        configuration=None,
-        credentials=None,
-        use_bqstorage_api=None,
-        private_key=None,
-        verbose=None,
-        progress_bar_type=None,
-        max_results=None,
-    ):  # noqa: PR01
+    def read_gbq(cls, query: str, **kwargs):  # noqa: PR01
```
Comment (vnlitvinov) [inline modin/core/io/io.py:177]: I don't like that you're changing API in a PR marked as `TEST-`.

Plus this change hurts IDE hinting and interactive help...

File: modin/pandas/io.py [hunk 6]
```diff
@@ -716,17 +716,17 @@ def to_pickle(
 @enable_logging
 def read_spss(
     path: Union[str, pathlib.Path],
-    usecols: Union[Sequence[str], type(None)] = None,
+    usecols: Sequence[str] | None = None,
```
Comment (vnlitvinov) [inline modin/pandas/io.py:719]: This is Python 3.10 syntax, and our minimum version is 3.8. I suggest using `Optional`:

```suggestion
    usecols: Optional[Sequence[str]] = None,
```

File: modin/pandas/io.py [hunk 7]
```diff
@@ -716,17 +716,17 @@ def to_pickle(
...
     convert_categoricals: bool = True,
 ):  # noqa: PR01, RT01, D200
     """
     Load an SPSS file from the file path, returning a DataFrame.
     """
+    _, _, _, kwargs = inspect.getargvalues(inspect.currentframe())
+
```
Comment (vnlitvinov) [inline modin/pandas/io.py:726]: let's not introduce this line... it's kinda slow and not that much useful when we have only 3 arguments

File: modin/pandas/io.py [hunk 8]
```diff
@@ -716,17 +716,17 @@ def to_pickle(
...
     from modin.core.execution.dispatching.factories.dispatcher import FactoryDispatcher
 
-    return DataFrame(
-        query_compiler=FactoryDispatcher.read_spss(path, usecols, convert_categoricals)
-    )
+    return DataFrame(query_compiler=FactoryDispatcher.read_spss(**kwargs))
```
Comment (vnlitvinov) [inline modin/pandas/io.py:729]: I would rather we pass explicitly, it's only three arguments to spell, not ~19 like in `.read_csv`

File: modin/pandas/test/test_io.py [hunk 9]
```diff
@@ -2491,6 +2493,20 @@ def test_to_gbq(self):
...
             modin_df.to_gbq("modin.table")
 
+    def test_read_gbq_mock(self):
+        test_args = ("fake_query",)
+        test_kwargs = {"project_id": "test_id", "dialect": "standart"}
+        all_params = inspect.signature(pd.read_gbq).parameters
+        for param in all_params:
+            # 'query' - positional arg
+            if param not in test_kwargs and param != "query":
+                test_kwargs[param] = all_params[param].default
```
Comment (vnlitvinov) [inline modin/pandas/test/test_io.py:2503]: maybe something like (untested)
```suggestion
        test_kwargs = inspect.signature(pd.read_gbq).parameters.copy()
        test_kwargs.update(project_id="test_id", dialect="standart")
        test_kwargs.pop("query", None)
```
?

File: modin/pandas/test/test_io.py (hunk 1 above)
Comment (vnlitvinov) [reply modin/pandas/test/test_io.py:2651]: this should be in-code comment 😉

File: modin/pandas/io.py (hunk 6 above)
Comment (anmyachev) [reply modin/pandas/io.py:719]: You are right, but there are many such places in our code. Something will need to be done about this as well.

File: modin/pandas/test/test_io.py (hunk 1 above)
Comment (anmyachev) [reply modin/pandas/test/test_io.py:2648]: added

File: modin/core/io/io.py (hunk 5 above)
Comment (anmyachev) [reply modin/core/io/io.py:177]: reverted

**Comment by anmyachev** [comment]
@vnlitvinov thanks for review! Ready for another one :)

**Comment by anmyachev** [comment]
@vnlitvinov ping

**Review by vnlitvinov — APPROVED** [review]
LGTM!

**Pull request merged**